driver-reactive-streams/src/test/functional/com/mongodb/reactivestreams/client/syncadapter/SingleResultSubscriber.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright 2008-present MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.mongodb.reactivestreams.client.syncadapter;
+
+import com.mongodb.MongoInterruptedException;
+import com.mongodb.MongoTimeoutException;
+import com.mongodb.lang.Nullable;
+import org.reactivestreams.Subscriber;
+import org.reactivestreams.Subscription;
+
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+
+class SingleResultSubscriber<T> implements Subscriber<T> {
+    private volatile T result;
+    private volatile Throwable exception;
+    private final CountDownLatch latch = new CountDownLatch(1);
+
+    @Nullable
+    T get() {
+        try {
+            if (!latch.await(30, TimeUnit.SECONDS)) {
```

Comment:
This timeout is so high in order to work around the async server selection bug I mentioned today.